### PR TITLE
README.FreeBSD: Status update 20170714

### PR DESCRIPTION
**Pull request by wjwithagen**
Signed-off-by: Willem Jan Withagen <wjw@digiware.nl>

### CODE REVIEW

File: README.FreeBSD [hunk 1]
```diff
@@ -11,152 +11,112 @@ In the end the cluster would be running bhyve on RBD disk that are stored in
...
 Build Prerequisites
 ===================
 
-	Compiling and building Ceph is tested on 12-CURRENT, but I guess/expect 
-	11-RELEASE will also work. And Clang is at 3.8.0.
-	It uses the CLANG toolset that is available, 3.7 is no longer tested, 
-	but was working when that was with 11-CURRENT. 
-	Clang 3.4 (on 10.2-STABLE) does not have all required capabilites to 
-	compile everything
+   Compiling and building Ceph is tested on 12-CURRENTclang 4.0.0, i
```
Comment (tchaikov) [inline README.FreeBSD]: add a slash after CURRENT.

File: README.FreeBSD [hunk 2]
```diff
@@ -11,152 +11,112 @@ In the end the cluster would be running bhyve on RBD disk that are stored in
...
+   Work on ceph-disk and ceph-deploy to make it more FreeBSD and ZFS 
+   compatible. 
+
+ - Build test-cluster and start running some of the teuthology integration 
+   tests on these.
+   Teuthology want to build its own libvirt and that does not quite work
+   with all the packages FreeBSD already has in place. Lots of minute 
+   details to figure out
+
+ - Design a vitual disk implementation that can be used with behyve and
```
Comment (tchaikov) [inline README.FreeBSD]: virtual

File: README.FreeBSD [hunk 3]
```diff
@@ -11,152 +11,112 @@ In the end the cluster would be running bhyve on RBD disk that are stored in
...
-	Teuthology want to build its own libvirt and that does not quite work
-	with all the packages FreeBSD already has in place. Lots of minute 
-	details to figure out
-
- -	Design a vitual disk implementation that can be used with behyve and 
-	attached to an RBD image. 
+ - Build an automated test platform that will build ceph/master on 
+   FreeBSD and report the results back to the Ceph developers. This will 
+   increase the maintainability of the FreeBSD side of things. 
+   Developers are signalled that they are using Linux-isms that will not
```
Comment (tchaikov) [inline README.FreeBSD]: signaled

**Comment by tchaikov** [comment]
@wjwithagen ping

**Comment by wjwithagen** [comment]
@tchaikov 
Pong!

File: README.FreeBSD (hunk 1 above)
Comment (tchaikov) [inline README.FreeBSD]: @wjwithagen  here?

File: README.FreeBSD [hunk 4]
```diff
@@ -11,152 +11,112 @@ In the end the cluster would be running bhyve on RBD disk that are stored in
...
 ===================
 
-	Compiling and building Ceph is tested on 12-CURRENT, but I guess/expect 
-	11-RELEASE will also work. And Clang is at 3.8.0.
-	It uses the CLANG toolset that is available, 3.7 is no longer tested, 
-	but was working when that was with 11-CURRENT. 
-	Clang 3.4 (on 10.2-STABLE) does not have all required capabilites to 
-	compile everything
+   Compiling and building Ceph is tested on 12-CURRENTclang 4.0.0, i
+   but I 11-RELEASE/clang 3.8.0.
```
Comment (tchaikov) [inline README.FreeBSD]: I what?

File: README.FreeBSD (hunk 3 above)
Comment (tchaikov) [inline README.FreeBSD]: @wjwithagen here?

File: README.FreeBSD (hunk 4 above)
Comment (wjwithagen) [inline README.FreeBSD]: @tchaikov 

That paragraph was a mess in the wip. Took a fresh one from the FreeBSD reporting.
Should be more legible now.

File: README.FreeBSD (hunk 3 above)
Comment (tchaikov) [inline README.FreeBSD]: @wjwithagen again?

File: README.FreeBSD [hunk 5]
```diff
@@ -11,152 +11,110 @@ In the end the cluster would be running bhyve on RBD disk that are stored in
...
+   in: ./src/common/Thread.cc
+
+ - Improve the FreeBSD /etc/rc.d initscripts in the Ceph stack. Both 
+   for testing, but mainly for running Ceph on production machines.
+   Work on ceph-disk and ceph-deploy to make it more FreeBSD and ZFS 
+   compatible. 
+
+ - Build test-cluster and start running some of the teuthology integration 
+   tests on these.
+   Teuthology want to build its own libvirt and that does not quite work
```
Comment (tchaikov) [inline README.FreeBSD]: wants.

File: README.FreeBSD [hunk 6]
```diff
@@ -11,152 +11,110 @@ In the end the cluster would be running bhyve on RBD disk that are stored in
...
-The following setup will get things running for FreeBSD:
-	
-	This all require root privilidges.
+   The following will get things running for FreeBSD:
+   (It requires root priviledges.)
 
- -	Install bash and link it in /bin
-	sudo pkg install bash
-	sudo ln -s /usr/local/bin/bash /bin/bash
+ - Install bash and link it in /bin
```
Comment (amitkumar50) [inline README.FreeBSD]: @wjwithagen Looks spaces left

**Comment by alfredodeza** [comment]
jenkins test docs

**Comment by alfredodeza** [comment]
@wjwithagen is this file meant to be `rst` ? If it is out of the doc/ dir it will not get built. Maybe that is OK for your purposes?

File: README.FreeBSD [hunk 7]
```diff
@@ -11,152 +11,112 @@ In the end the cluster would be running bhyve on RBD disk that are stored in
...
-Most important change:
-  -	A port is submitted: net/ceph-devel.
+     
+Most important changes:
+ - Ceph has releassed the release candidate of v12.1.0 (aka Luminous),
+   the port is sitting in my tree waiting for Luminous to be actually 
+   released.
+ - ceph-fuse works, and allows mounting of cephfs filesystems. 
+   Speed is not impressive, but is does work.
+ - rbd-ggate available to create a Ceph rdb backed device.
```
Comment (wjwithagen) [inline README.FreeBSD]: @amitkumar50 
According to the Oxford dictionary is 'back' a verb, so backed would be correct grammar.
But I'll fix it.

**Comment by wjwithagen** [comment]
@alfredodeza 
It is more to match the other README.<platform> files that are in Ceph root.
There is also a `doc/dev/freebsd.rst`

**Comment by liewegas** [comment]
@wjwithagen does this need an update or is it obsolete?

**Comment by wjwithagen** [comment]
@liewegas 
I think this needs updating.
I keep it as HEAD notices for others that want to work on Ceph, instead of installing a package.

**Comment by wjwithagen** [comment]
Update needs to be reworked since the FreeBSD ceph package release.

**Pull request closed without merging**